pkg/termio/ask_test.go

```diff
@@ -214,25 +214,30 @@ func TestAskForPasswordNonInteractive(t *testing.T) {
 	ctx := context.Background()
 	ctx = ctxutil.WithInteractive(ctx, false)
 
-	_, err := AskForPassword(ctx, "test")
+	_, err := AskForPassword(ctx, "test", true)
 	assert.Error(t, err)
 
 	// provide value on redirected stdin
 	input := `foo
 foo
 foobar
 foobaz
+foobat
 `
 
 	Stdin = strings.NewReader(input)
 	ctx = ctxutil.WithAlwaysYes(ctx, false)
 	ctx = ctxutil.WithInteractive(ctx, true)
 	ctx = ctxutil.WithTerminal(ctx, false)
-	sv, err := AskForPassword(ctx, "test")
+	sv, err := AskForPassword(ctx, "test", true)
 	assert.NoError(t, err)
 	assert.Equal(t, "foo", sv)
 
-	sv, err = AskForPassword(ctx, "test")
+	sv, err = AskForPassword(ctx, "test", false)
+	assert.NoError(t, err)
+	assert.Equal(t, "foobar", sv)
+
+	sv, err = AskForPassword(ctx, "test", true)
 	assert.NoError(t, err)
```

Comment:
Is this really the behaviour we want, btw, not erroring out but instead outputting the empty string is the two passwords aren't matching?

Reply:
I don't know, but that's the original functionality and isn't changing here, so it seems outside the scope of this PR.

Reply:
Agreed, but I was raising this for discussion with @dominikschulz to see if we needed to address this in another PR